packages/protocol/contracts/common/UsingPrecompiles.sol

```diff
@@ -16,6 +18,28 @@ contract UsingPrecompiles {
   address constant GET_PARENT_SEAL_BITMAP = address(0xff - 10);
   address constant GET_VERIFIED_SEAL_BITMAP = address(0xff - 11);
 
+  /**
+   * @notice calculate a * b^x
+   * @param a First unwrapped Fraction.
+   * @param b The unwrapped Fraction to be exponentiated.
+   * @param exponent Exponent to raise b to.
+   * @return The computed quantity as an unwrapped Fraction.
+   */
+  function mulExp(uint256 a, uint256 b, uint256 exponent) public view returns (uint256) {
+    uint256 denominator = FixidityLib.fixed1().unwrap();
+    uint256 returnNumerator;
+    uint256 returnDenominator;
+    (returnNumerator, returnDenominator) = fractionMulExp(
```

Comment:
Ideally what we want is for the precompile to actually use fixidity numbers and thus receive only 3 parameters. But this is a good first step.

The problem is that to make that change, we need to:

 1. change geth
 2. change ganache/ethereumvm

But it would be great to do this before mainnet.... after that it will be harder

Reply:
How important is saving 64 bytes of argument packing and unpacking?  If we decide to keep `StableToken.digits_`, then it might be best to have two precompiles.

Reply:
+1, I was imagining we would change the precompiles here as well, I'm not sure of the value of this PR in a vacuum